aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -160,6 +160,21 @@ static inline int64_t matrixStride(const Tensor& batched_matrices) {
   return batched_matrices.size(-1) * batched_matrices.size(-2);
 }
 
+// This function is similar to at::native::toListOfOptionalTensors,
+// but converts inputs to at::indexing::TensorIndex and outputs a std::vector.
+// The output could then be supplied with at::indexing::Ellipsis or
+// at::indexing::Slice, for example, to perform advanced indexing
+// with the `index` method.
+static inline std::vector<at::indexing::TensorIndex>
+toVectorOfTensorIndices(ArrayRef<Tensor> v, c10::optional<int64_t> len = c10::nullopt) {
+  std::vector<at::indexing::TensorIndex> result;
+  result.reserve(len.has_value() ? len.value() : v.size());
+  for (const Tensor& t : v) {
+    result.push_back(t);
+  }
```

Comment:
@peterbell10 , the issue is here with `TensorIndex` being incomplete.

Reply:
`TensorIndexing.h` is not `TORCH_ASSERT_ONLY_METHOD_OPERATORS` compliant but `LinearAlgrebraUtils.h` is used in contexts where the macro is defined so you can't use `TensorIndex` in this header.  I would suggest just using `vector::assign`. It's still a one-liner and handles the implicit conversion just the same.

```
      rank_r_indices.assign(rank_r_mask.begin(), rank_r_mask.end());
```